### DISCUSSION TITLE
Ignored Build Step and main branch

### DISCUSSION THREAD

**Discussion by sztadii** [vercel/vercel#8410] [Help] [answered]
Hi, is it possible to have access to the main branch?
I have a repo where I need to compare changes with the main branch, but it is not accessible.

My script looks like this
```
#!/bin/bash

# add an origin for the git repository
git remote add origin git@github.com:sztadii/monorepo.git

git fetch origin main

# get the project_root path from arguments
project_root=$1
package_name=$2
changes=$(npx yarn-changed-workspaces@2.0.11 --projectRoot "$project_root" --branch origin/main)

if [[ "$changes" == *"$package_name"* ]] ; then
  echo "✅ - Build can proceed"
  exit 1;

else
  echo "🛑 - Build canceled"
  exit 0;
fi

```

We are running the `yarn-changed-workspaces` package to see if the workspace has any new changes ( or their latest deps have any changes ), but for that, I need a second branch to compare. Any ideas on how to do this?

**Comment by amyegan** [comment, marked as answer]
Hi @sztadii! It looks like you already read our [support article](https://vercel.com/support/articles/how-do-i-use-the-ignored-build-step-field-on-vercel) and [Ignored Build Step](https://vercel.com/docs/concepts/projects/overview#ignored-build-step) documentation. Changes can typically be detected based on comparison of the new commit and previous commits in the branch. And you can [test commands locally](https://vercel.com/support/articles/how-do-i-use-the-ignored-build-step-field-on-vercel#debugging-commands-locally) to try out your script or [git diff](https://git-scm.com/docs/git-diff) command before adding it to your project settings. 

If you're deploying a monorepo using Yarn Workspaces, I would recommend checking out our article about [Deploying a Monorepo Using Yarn Workspaces to Vercel](https://vercel.com/guides/deploying-yarn-monorepos-to-vercel#configure-the-ignored-build-step-on-vercel). There's a section on [configuring the Ignored Build Step](https://vercel.com/guides/deploying-yarn-monorepos-to-vercel#configure-the-ignored-build-step-on-vercel) that might be what you need.

**Reply by jdpnielsen** [reply]
Is there any way to configure Vercel to fetch changes for other branches than the current one? Checking only in comparison to the current branch (and the [10 latest commits](https://vercel.com/guides/how-do-i-use-the-ignored-build-step-field-on-vercel#debugging-commands-locally)) is fairly limiting.

**Comment by sztadii** [comment]
The problem with git diff is not checking the related deps.
For example 
the `package2` is used by the `package1`, but `git diff` will only check if the current package has any changes, but not their dependencies.
```"package_name": "workspace:^",```

**Comment by sztadii** [comment]
That's why `yarn-changed-workspaces` is soo helpful because their script in an easy way is checking if the workspace is affected.
